microsoft/testsuites/gpu/gpusuite.py

```diff
@@ -0,0 +1,138 @@
+            "Standard_ND" in node_runbook.vm_size
+            or "Standard_NV" in node_runbook.vm_size
+        ) and "v3" in node_runbook.vm_size:
+            expected_count = math.ceil(core_count / 12)
+
+        return expected_count
+
+    @TestCaseMetadata(
+        description="""
+            This test case verfies kernel support for gpu.
+            TODO: validate the test params against GPU support test matrix
+                    when test matrix is available.
+        """,
+        requirement=simple_requirement(
+            supported_features=[Gpu],
+        ),
+    )
+    def validate_gpu_support(self, case_name: str, node: Node) -> None:
+        gpu_feature = node.features[Gpu]
+        gpu_feature.check_support()
+
+    @TestCaseMetadata(
+        description="""
+            This test case verfies if gpu drivers are loaded fine.
+
+            Steps:
+            1. Validate if the VM SKU is supported for GPU.
+            2. Install LIS drivers if not already installed for Fedora and its
+                derived distros. Reboot the node
+            3. Install required gpu drivers on the VM and reboot the node. Validate gpu
+                drivers can be loaded successfully.
+
+        """,
+        requirement=simple_requirement(
+            supported_features=[Gpu, SerialConsole],
+        ),
+    )
+    def validate_load_driver(self, case_name: str, node: Node) -> None:
+        gpu_feature = node.features[Gpu]
+        node_runbook = node.capability.get_extended_runbook(AzureNodeSchema, AZURE)
+        case_path = self._create_case_log_path(case_name)
+
+        if isinstance(node.os, Redhat):
+            # install LIS driver if not already installed.
+            lisdriver = node.tools[LisDriver]
+            if not lisdriver.get_version():
+                lisdriver.install_driver()
+                self._validate_node_reboot(node, case_path)
+
+        node_runbook = node.capability.get_extended_runbook(AzureNodeSchema, AZURE)
+        # Install required compute SDKs and then reboot to load the drivers.
+        if any(map((node_runbook.vm_size).__contains__, self.grid_supported_skus)):
+            self.log.debug(
+                f"The VM {node_runbook.vm_size}"
+                f" instance is with GRID device driver."
+            )
+            gpu_feature.install_compute_sdk(ComputeSDK.GRID)
+        elif any(map((node_runbook.vm_size).__contains__, self.cuda_supported_skus)):
+            self.log.debug(
+                f"The VM {node_runbook.vm_size}"
+                f" instance is with CUDA device driver."
+            )
+            gpu_feature.install_compute_sdk(ComputeSDK.CUDA)
+        else:
+            raise SkippedException("Azure VM size not supported for GPU.")
+
+        self._validate_node_reboot(node, case_path)
+
+        # TODO - More logic to be added if support for more drivers added.
+        lsmod_tool = node.tools[Lsmod]
+        if not lsmod_tool.module_exists("nvidia"):
```

Comment:
Because it needs to detect nvidia or cuda, move it to feature to add a method to validate the driver is installed successfully. like `is_driver_installed`